packages/flutter_test/lib/src/controller.dart

```diff
@@ -991,6 +1019,48 @@ abstract class WidgetController {
     return tapAt(getCenter(finder, warnIfMissed: warnIfMissed, callee: 'tap'), pointer: pointer, buttons: buttons);
   }
 
+  /// Dispatch a pointer down / pointer up sequence at the first hit-testable
+  /// [InlineSpan] (typically a [TextSpan]) within the given text range.
+  ///
+  /// This method performs a more spatially precise tap action on a piece of
+  /// static text, than the widget-based [tap] method.
+  ///
+  /// The given [Finder] must find one and only one matching substring, and the
+  /// substring must be hit-testable (meaning, it must not be off-screen, or be
+  /// obscured by other widgets, or in a disabled widget). Otherwise this method
+  /// throws a [FlutterError].
+  ///
+  /// If the target substring contains more than one hit-testable [InlineSpan]s,
+  /// [tapOnText] taps on one of them, but does not guarantee which.
```

Comment:
Huh? The first paragraph said it would hit the first one. This one says it doesn't guarantee which one. Can you unify these two statements? Also, if the "no guarantee" statement is the correct one: Why can we not guarantee which one gets hit? If we cannot define which one gets hit, should we instead throw in this case and make that illegal? Is it at least deterministic which gets hit?